test/specs/integration.spec.js

```diff
@@ -65,125 +62,116 @@ describe('Integration tests', function() {
         cm1.replaceRange('', {line: 0, ch: 10}, {line: 0, ch: 13});
         cm1.replaceRange('', {line: 0, ch: 0},  {line: 0, ch: 3});
       });
+      setTimeout(function() {
+        expect(cm2.getValue()).toEqual('3456789');
+        done();
+      }, 500);
     });
-    waitsFor(function() { return cm2.getValue() === '3456789' }, 'Got correct end value.');
   });
 
-  it('Random text changes', function () {
+  it('Random text changes', function (done) {
     var ref = new Firebase('https://firepad-test.firebaseio-demo.com').push();
-    var cm1 = CodeMirror(cmDiv());
-    var cm2 = CodeMirror(cmDiv());
+    var cm1 = CodeMirror(hiddenDiv());
+    var cm2 = CodeMirror(hiddenDiv());
     var firepad1 = new Firepad(ref, cm1);
     var firepad2 = new Firepad(ref, cm2);
 
-    var n = 25;
-
-    var done = false;
-    function step () {
-      if (n == 0) {
-        done = true;
-        return;
+    function step (times) {
+      if (times == 0) {
+        done();
+      }else{
+        randomOperation(cm1);
+        setTimeout(function() {
```

Comment:
I can, but the test now consistently passes with a `console.log` in there, and fails without it.
```js

  it('Random text changes', function(done) {
    var ref = new Firebase('https://firepad-test.firebaseio-demo.com').push();
    var cm1 = CodeMirror(hiddenDiv());
    var cm2 = CodeMirror(hiddenDiv());
    var firepad1 = new Firepad(ref, cm1);
    var firepad2 = new Firepad(ref, cm2);

    function step(times) {
      if (times == 0) {
        expect(cm1.getValue()).toEqual(cm2.getValue());
        done();
      } else {
        randomOperation(cm1);
        var onchange = function() {
          console.log("change"); // secret magic
          if (cm1.getValue() === cm2.getValue()) {
            cm2.off('change', onchange);
            step(times - 1);
          }
        };
        cm2.on('change', onchange);
      }
    }

    firepad1.on('ready', function() {
      firepad1.setText('lorem ipsum');
      step(25);
    });
  }, 1000 * 25 + 1000);
```